hasjob/static/js/app.js

```diff
@@ -184,6 +184,199 @@ window.Hasjob.PaySlider.prototype.resetSlider = function(currency) {
+  };
+
+  var paySlider = new Hasjob.PaySlider({
+    start: query['pmin'] || 0,
+    end: query['pmax'] || 10000000,
+    selector: "#pay-slider",
+    minField: '#job-filters-pmin',
+    maxField: '#job-filters-pmax'
+  });
+
+  $("#pay-slider").on('slide', function(){
+    setPayTextField();
+  });
+
+  setPaySliderVisibility();
+  paySlider.resetSlider(getCurrencyVal());
+  setPayTextField();
+
+  //remove white spaces keyword input value
+  $("#job-filters-keywords").on("change",function(){
+    $(this).val($(this).val().trim());
+  });
+
+  $('#job-filters').on('submit', function(e){
+    // remove currency params from URL if currency is n/a
+    e.preventDefault();
+    var formParams = $(this).serializeArray();
+    var sortedFilterParams = [];
+    for (var fpIndex=0; fpIndex < formParams.length; fpIndex++) {
+      // set value to empty string if currency is n/a
+      if (formParams[fpIndex].name === 'currency') {
+        if (formParams[fpIndex].value.toLowerCase() === 'na') {
+          formParams[fpIndex].value = "";
+        }
+        var currencyVal = formParams[fpIndex].value;
+      }
+      // format pmin and pmax based on currency value
+      if (formParams[fpIndex].name === 'pmin' || formParams[fpIndex].name === 'pmax') {
+        if (currencyVal === '') {
+          formParams[fpIndex].value = '';
+        } else {
+          formParams[fpIndex].value = Hasjob.PaySlider.toNumeric(formParams[fpIndex].value);
+        }
+      }
+      // remove empty values
+      if (formParams[fpIndex].value !== '') {
+        sortedFilterParams.push(formParams[fpIndex]);
+      }
+    }
+
+    // only redirect if there are filters applied
+    if (sortedFilterParams.length > 0) {
+      window.location.href = '/'+ '?' + $.param(sortedFilterParams);
+    } else {
+      window.location.href = '/';
+    }
+  });
+  
+  $("#job-filters-location").multiselect({
+    nonSelectedText: 'Location',
+    numberDisplayed: 1,
+    buttonWidth: '100%',
+    enableFiltering: true,
+    enableCaseInsensitiveFiltering: true,
+    templates: {
+      filter: '<li><div class="input-group input-group-sm"><div class="input-group-addon"><i class="fa fa-search"></i></div><input type="text" class="form-control" id="job-filter-location-search" placeholder="Search">',
+      filterClearBtn: '<div class="input-group-addon job-filter-location-search-clear"><i class="fa fa-times"></i></div></div></li>'
+    }
+  });
+
+  // clear location search on clicking the clear control
+  $('.job-filter-location-search-clear').click(function(e){
+    $('#job-filter-location-search').val('');
+  });
+
+  $("#job-filters-type").multiselect({
+    nonSelectedText: 'Job Type',
+    numberDisplayed: 1,
+    buttonWidth: '100%'
+  });
+
+  $("#job-filters-category").multiselect({
+    nonSelectedText: 'Job Category',
+    numberDisplayed: 1,
+    buttonWidth: '100%'
+  });
+});
+
+$(document).keydown(function(event) { 
+  if (event.keyCode==27 && $("#hg-sitenav").hasClass("in")){
```

Comment:
- Almost always, please use `===` instead of `==` in JS. The latter can causing confusing behavior. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness
- Also, add spacing around the `===`.
- I don't think you need to return false here. Also, good practice to put `event.preventDefault() right at the top.